packages/client/src/components/app/charts/ApexChart.svelte

```diff
@@ -1,25 +1,82 @@
 <script>
   import { getContext } from "svelte"
-  import { chart } from "svelte-apexcharts"
-  import Placeholder from "../Placeholder.svelte"
+  import ApexCharts from "apexcharts"
+  import { Icon } from "@budibase/bbui"
+  import { cloneDeep } from "lodash"
 
   const { styleable, builderStore } = getContext("sdk")
   const component = getContext("component")
 
   export let options
+
+  // Apex charts directly modifies the options object with default properties and internal variables. These being present could unintentionally cause issues to the provider of this prop as the changes are reflected in that component as well. To prevent any issues we clone options here to provide a buffer.
+  $: optionsCopy = cloneDeep(options)
+
+  let chartElement
+  let chart
+  let currentType = null
+
+  const updateChart = async newOptions => {
+    // Line charts have issues transitioning between "datetime" and "category" types, and will ignore the provided formatters
+    // in certain scenarios. Rerendering the chart when the user changes label type fixes this, but unfortunately it does
+    // cause a little bit of jankiness with animations.
+    if (newOptions?.xaxis?.type && newOptions.xaxis.type !== currentType) {
```

Comment:
Optional chaining on the right hand side also? 

```javascript
newOptions?.xaxis?.type !== currentType
```

Reply:
If the first condition passes then the optional chaining won't be necessary I think.